src/main/java/com/example/springbootbookshop/service/impl/UserServiceImpl.java

```diff
@@ -26,11 +27,10 @@ public class UserServiceImpl implements UserService {
     private final PasswordEncoder passwordEncoder;
 
     @Override
-    public UserResponseDto register(UserRegistrationRequestDto requestDto)
-            throws RegistrationException {
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         if (userRepository.existsUserByEmail(requestDto.email())) {
-            throw new RegistrationException("User with email: " + requestDto.email()
-            + " already exist");
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
```

Comment:
what is the point of this change? why `RegistrationException` isn't suitable?

Reply:
Lilia said it's better to return 4xx status instead of 5xx

Reply:
@don-bigdad then you misunderstood her :)

it is okay to throw `RegistrationException`

you just need to handle it properly in `@ControllerAdvice`:
```
@ExceptionHandler(RegistrationException.class)
// return error with HTTP 409 CONFLICT
```